### PR TITLE
Fix memory leaks caused by a duplicated message creation

### PR DESCRIPTION
**Pull request by yuliasherman**
* **Problem Overview**  
- After introducing the unique pointers in the messageHandler and onMessage APIs new memory leaks appeared. The problem was caused by a missing delete of an original message pointer.
- In addition, some unused functions have been removed.
- An initial call to the validate functions in the PreProcessor code has been fixed to call the correct virtual ones.
* **Testing Done**  
Apollo tests with asan (skvbc_basic_tests_v4, test_skvbc_reconfiguration, test_skvbc_state_transfer, skvbc_preexecution_tests), long-run Maestro tests with interruptions.

### CODE REVIEW

File: bftengine/src/bftengine/ReadOnlyReplica.hpp [hunk 1]
```diff
@@ -43,7 +43,8 @@ class ReadOnlyReplica : public ReplicaForStateTransfer {
 
   template <typename T>
   void messageHandler(std::unique_ptr<MessageBase> msg) {
-    auto trueTypeObj = std::make_unique<T>(msg.release());
+    auto trueTypeObj = std::make_unique<T>(msg.get());
+    delete msg.release();
```
Comment (glevkovich) [inline bftengine/src/bftengine/ReadOnlyReplica.hpp:47]: The proper way to do this is:
`msg.reset()`
Same for all other places where you
`delete smart_ptr.release();`
(I won't comment anywhere else)

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 2]
```diff
@@ -252,7 +253,7 @@ void ReplicaImp::validatedMessageHandler(CarrierMesssage *msg) {
     }
   }
   if (!isCollectingState()) {
-    onMessage<T>(std::make_unique<T>(trueTypeObj));
+    onMessage<T>(std::unique_ptr<T>(trueTypeObj));
```
Comment (glevkovich) [inline bftengine/src/bftengine/ReplicaImp.cpp:256]: nit: just carious why did you change this all over the code?
In general, std::make_unique is safer during compile time (wil lgenerate more errors if you did something wrong).

File: utt-replica/signature-processor/src/sigProcessor.cpp [hunk 3]
```diff
@@ -88,7 +88,7 @@ SigProcessor::SigProcessor(uint16_t repId,
       timer_handler_timeout_{timer_handler_timeout} {
   msg_handlers->registerMsgHandler(
       bftEngine::impl::MsgCode::Reserved, [&](std::unique_ptr<bftEngine::impl::MessageBase> message) {
-        auto msg = std::make_unique<messages::PartialSigMsg>(message.release());
+        std::unique_ptr<messages::PartialSigMsg> msg((messages::PartialSigMsg*)message.release());
```
Comment (glevkovich) [inline utt-replica/signature-processor/src/sigProcessor.cpp:91]: `reinterpret_cast<messages::PartialSigMsg*>(message.release())`
same for line 297

**Review by glevkovich — APPROVED** [review]
@yuliasherman 
Approved, Thanks, with few small comments/questions.

**Review by toly-kournik — APPROVED** [review]
LGTM

File: bftengine/src/bftengine/ReadOnlyReplica.hpp (hunk 1 above)
Comment (yuliasherman) [reply bftengine/src/bftengine/ReadOnlyReplica.hpp]: Agree

File: utt-replica/signature-processor/src/sigProcessor.cpp (hunk 3 above)
Comment (yuliasherman) [reply utt-replica/signature-processor/src/sigProcessor.cpp]: OK

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 2 above)
Comment (yuliasherman) [reply bftengine/src/bftengine/ReplicaImp.cpp:256]: make_unique calls a copy constructor and creates a new object of type "T", which causes a memory leak.
